py3status/modules/gazelle.py

```
@@ -0,0 +1,164 @@
+            self._index()  # just to try if login info are good
+        except RequestException:
+            self.py3.log(
+                    "Cannot login to %s" % self.baseurl,
+                    level=self.py3.LOG_ERROR
+                    )
+            raise
+        self._cookiejar.save(self.baseurl)
+        self.py3.log(
+                "Saved cookie in storage",
+                level=self.py3.LOG_INFO
+                )
+
+    def _index(self):
+        resp = self.py3.request(
+                self.baseurl+"/ajax.php",
+                params={'action': "index"},
+                headers={
+                    'User-Agent': 'Mozilla/5.0'
+                    },
+                cookiejar=self._cookiejar
+                )
+        return resp.json()
+
+    def post_config_hook(self):
+        # Check if given URL is really an URL
+        if not isinstance(self.baseurl, str) or not re.search("^https?://", self.baseurl):
+            self.py3.error("Invalid URL")
+        self.baseurl = self.baseurl.strip('/')
+        if not isinstance(self.username, str) or not isinstance(self.password, str):
+            self.py3.error("Invalid login info")
+        # Make a copy of thresholds list with keywords
+        self._thresholds_orig = list(self.thresholds)
+
+        self._cookiejar = self.py3.storage_new_cookiejar()
+        try:
+            self._cookiejar.load(self.baseurl)
+            self._index()  # just to try if saved cookie is good
+        except RequestException:
+            self.py3.log(
+                    "Cookie invalid, outdated or missing. Re-logging.",
+                    level=self.py3.LOG_INFO
+                    )
+            self._login()
+
+    def gazelle(self):
+        try:
+            index = self._index()
+            if index['status'] != "success":
+                msg = "Tracker failed to give me user info"
+                self.py3.log(msg, level=self.py3.LOG_ERROR)
+                raise RequestException(msg)
```

Does this trigger when users have no Internet? If yes, maybe this is a bit harsh... Better behavior, I think, would be to fail silently and keep trying... Earlier review indicates that this can fail. `index.get('status')` may be a safe move.

---

That's why we nee a standard way to treat offline modules. I know this won't happen for now, so I'll disable this error raise.